src/Symfony/Component/HttpFoundation/Tests/Session/Storage/NativeSessionStorageTest.php

```diff
@@ -119,6 +119,19 @@ public function testRegenerateDestroy()
         $this->assertEquals(11, $storage->getBag('attributes')->get('legs'));
     }
 
+    public function testSessionGlobalIsUpToDateAfterIdRegeneration()
+    {
+        $storage = $this->getStorage();
+        $storage->start();
+        $storage->getBag('attributes')->set('lucky', 7);
+        $storage->regenerate();
+        $storage->getBag('attributes')->set('lucky', 42);
+
+        $this->assertArrayHasKey('_sf2_attributes', $_SESSION);
+        $this->assertArrayHasKey('lucky', $_SESSION['_sf2_attributes']);
```

Comment:
Do we really need these two tests? The test would fail anyway if the keys didn't exist.